netweaver/extract_nw_archives.sls

```diff
@@ -0,0 +1,51 @@
+{%- from "netweaver/map.jinja" import netweaver with context -%}
+{% set nw_extract_dir = netweaver.nw_extract_dir %}
+{% set swpm_extract_dir = nw_extract_dir~'/SWPM' %}
+
+setup_nw_extract_directory:
+  file.directory:
+    - name: {{ nw_extract_dir }}
+    - mode: 755
+    - makedirs: True
+
+{% if netweaver.sapcar_exe_file is defined and netweaver.swpm_sar_file is defined %}
+
+extract_installer_file:
+    sapcar.extracted:
+    - name: {{ netweaver.swpm_sar_file }}
+    - sapcar_exe: {{ netweaver.sapcar_exe_file }}
+    - output_dir: {{ nw_extract_dir }}/SWPM
+    - options: "-manifest SIGNATURE.SMF"
+
+{% endif %}
+
+{% set additional_dvd_folders = [] %}
+
+{% for dvd in netweaver.additional_dvds %}
+{% set dvd_folder = dvd.split('.')[0] %}
+{% if additional_dvd_folders.append(dvd_folder) %}{% endif %}
```

Comment:
This line doesn't make sense, you are running an operation in an `if`. This looks a bad practice for me in this scenario.
What do you want to do here? You could just do `{% do additional_dvd_folders.append(dvd_folder) %}` I guess

Reply:
In fact it's not only `dvd_folder`, it should be `nw_extract_dir~dvd_folder`, isn't it?

Reply:
I used the `{% do additional_dvd_folders.append(dvd_folder) %}` initially, but testing with jinja live parser gave me error using `do`. It seems we need to import the do expression extension. https://jinja.palletsprojects.com/en/master/extensions/#expression-statement

But since we already use it in our formulas and its working fine there I will try to test it now and if its working i will update